webclient/core/templates/core/export_modal.html

```diff
@@ -48,55 +45,60 @@ <h5 class="modal-title">{% trans "core.templates.core.export-modal.title" %}</h5
           fetch("{% url 'core:reset_temp_value' %}?name=export_{{ request.user.name }}")
               .catch(error => console.error('Error in reset function:', error));
       }
-
-      const enableCloseButton = () => {
-          const button = document.getElementById('export-progress-modal-button');
-          button.removeAttribute('disabled');
-          button.classList.remove('disabled');
+      
+      const controller = new AbortController();
+      const signal = controller.signal;
+      
+      const cancelButton = document.getElementById('export-progress-modal-button-cancel');
+      cancelButton.onclick = () => {
+          controller.abort();
       }
 
-      document.addEventListener('DOMContentLoaded', function (event) {
-          document.getElementById('export-progress-modal-button').addEventListener('click', resetTempValue);
-          try {
-            document.getElementById('export-progress-modal-button-cancel').addEventListener('click', resetTempValue);
-          }
-          catch {
-            console.log("no element export-progress-modal-button-cancel")
-          }
-          document.getElementById('export-progress-modal-button-stop').addEventListener('click', () => {
-              window.stop()
-              enableCloseButton()
-              resetTempValue()
-          })
-      })
+      const manageExportDownload = (target_link) => {
+          const downloadUrl = target_link;
+          
+          const downloadStartEvent = new Event('downloadStart');
+          document.dispatchEvent(downloadStartEvent);
 
-      const checkValue = () => {
-          const cancelButton = document.getElementById("export-progress-modal-button-cancel")
-          if (cancelButton){
-              cancelButton.style.display = "none";
-          }
-          const stopButton = document.getElementById("export-progress-modal-button-stop")
-          if (stopButton){
-              stopButton.style.display = null;
-          }
-          const button = document.getElementById("export-progress-modal-button")
-          if (button){
-              button.style.display = null;
-          }
           const rawEmail = "{{ request.user.email }}"
           const email = rawEmail.replace("@", "(at)")
           fetch(`{% url 'core:read_temp_value' %}?temp_name=export_${email}`)
               .then(response => response.json())
               .then(data => {
                   updateProgressBar(data.value);
-                  if (data.value !== 100) {
-                      setTimeout(checkValue, 1000);
-                  } else {
-                      enableCloseButton();
-                      resetTempValue();
-                  }
               })
               .catch(error => console.error('Error in Check Value Function:', error));
-      }
+
+          fetch(downloadUrl, { signal })
+              .then(response => {
+                  if (!response.ok) throw new Error('Network response was not ok.');
```

Comment:
@pesikj Tohle je hláška pro log, nebo pro uživatele? Pokud pro uživatele, měla by jít do překladů.

Reply:
To je hláška do logu, zobrazí se v konzoli (v developer tools - F12).